packages/grafana-ui/src/components/uPlot/plugins/TooltipPlugin2.tsx

```diff
@@ -340,6 +438,10 @@ export const TooltipPlugin2 = ({
         );
       }
 
+      if (queryZoom != null && getFeatureToggle('timeRangePan')) {
+        setupXAxisPan(u, queryZoom);
+      }
```

Comment:
📓 @leeoniya + @drew08t _(or any other reviewer)_:

I'm looking for feedback on whether or not we think `TooltipPlugin2` is the best place to implement this feature or not.

I did this because I wanted to have a handle on `queryZoom` to avoid conflict with existing click/drag interactions.

I'm open to alternative suggestions.